p2pclient/streams.go

```diff
@@ -0,0 +1,90 @@
+package p2pclient
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+
+	ggio "github.com/gogo/protobuf/io"
+	pb "github.com/libp2p/go-libp2p-daemon/pb"
+)
+
+// NewStream initializes a new stream on one of the protocols in protos with
+// the specified peer.
+func (c *Client) NewStream(peer []byte, protos []string) (*pb.StreamInfo, error) {
```

Comment:
I am confused here -- `NewStream` should create a new unix socket and return it together with the stream info; this hijacks the control socket.